app/src/main/java/org/mozilla/tv/firefox/webrender/cursor/CursorView.kt

```diff
@@ -5,89 +5,96 @@
 package org.mozilla.tv.firefox.webrender.cursor
 
 import android.content.Context
-import android.os.Handler
-import android.os.Looper
-import android.os.Message
-import androidx.annotation.UiThread
-import androidx.appcompat.widget.AppCompatImageView
+import android.graphics.Canvas
+import android.graphics.PointF
 import android.util.AttributeSet
-import android.view.KeyEvent
+import androidx.annotation.CheckResult
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.core.view.isVisible
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+import io.reactivex.rxkotlin.Observables
+import io.reactivex.rxkotlin.addTo
 import org.mozilla.tv.firefox.R
-import org.mozilla.tv.firefox.utils.RemoteKey
-import java.lang.ref.WeakReference
 import java.util.concurrent.TimeUnit
 
-private const val HIDE_MESSAGE_ID = 0
+private const val BITMAP_PRESSED = R.drawable.cursor_full_active
+private const val BITMAP_UNPRESSED = R.drawable.cursor_full
 private const val HIDE_ANIMATION_DURATION_MILLIS = 250L
 private val HIDE_AFTER_MILLIS = TimeUnit.SECONDS.toMillis(3)
 
 /**
- * A drawn Cursor: see [LegacyCursorViewModel] for responding to keys and setting position.
- * The cursor will hide itself when it hasn't received a location update recently.
+ * Renders the cursor visible on top of the browser.
+ *
+ * Most behavior is controlled by [CursorModel]
+ *
+ * Any images set must be Bitmaps, not Drawables
  */
 class CursorView(context: Context, attrs: AttributeSet) : AppCompatImageView(context, attrs) {
 
-    private val hideHandler = CursorHideHandler(this)
+    private val onDrawMutablePositionCache = PointF(x, y)
```

Comment:
nit: explain that this is only intended to be used in `onDraw` to avoid allocation and that it works because it's only ever called from the UI thread on the same method (no concurrent access)